python/cudf/cudf/core/dataframe.py

```diff
@@ -2969,25 +3058,61 @@ def drop(
             outdf = self
         else:
             outdf = self.copy()
-        for c in columns:
-            outdf._drop_column(c)
-        return outdf
 
-    def drop_column(self, name):
-        """Drop a column by *name*
-        """
-        warnings.warn(
-            "The drop_column method is deprecated. "
-            "Use the drop method instead.",
-            DeprecationWarning,
-        )
-        self._drop_column(name)
+        def _drop_columns(df, columns, errors):
+            for c in columns:
+                try:
+                    df._drop_column(c)
+                except KeyError as e:
+                    if errors == "ignore":
+                        pass
+                    else:
+                        raise e
+
+        if axis in (1, "columns"):
+            _drop_columns(outdf, target, errors)
+        elif axis in (0, "index"):
+            index = cudf.Series(target)
+            if isinstance(self._index, cudf.MultiIndex):
+                if level is None:
+                    level = 0
+
+                levels_index = outdf.index.get_level_values(level)
+                if errors == "raise" and not index.isin(levels_index).all():
+                    raise KeyError("One or more values not found in axis")
+
+                sliced_df = outdf.take(~levels_index.isin(target))
```

Comment:
Similarly we can use an anti-join here

Reply:
I think we cannot perform a join with multiIndex and a single values/1-d array?

Reply:
I vaguely remember there being a way to specify the level of a MultiIndex as a join key, but worst case you could pull the column out of the MultiIndex in order to do the join. Then it would be something like:
```
new_out_df = outdf.copy(deep=False)
new_out_df['join_col'] = new_out_df.index.get_level_values(level)
target_df = cudf.DataFrame({'join_col': target})
sliced_df  = outdf.merge(target_df, how="leftanti", on=["join_col"])
```

That being said this likely isn't order preserving which poses a problem.

Reply:
Lets just add a #TODO for now for future optimization.

Reply:
Sure